UsersData/Users/ViewModel/ViewModel.swift

```diff
@@ -13,33 +13,75 @@ enum Users {}
 
 extension Users  {
     @MainActor class ViewModel: ObservableObject {
-        var service: UserService?
+        
+        var cachedUsers:FetchedResults<User>?
+        var context:NSManagedObjectContext?
+        let service:NetworkServiceProtocol!
         
         @Published var error: String = ""
         @Published var isLoading: Bool = false
         @Published var errorIsPresented: Bool = false
+        @Published var users: [UserResponse] = []
+        
+        var isLast: Bool = false
+        var page = 0
+        var isRefresh = false
+        let amountPerPage = 15
         
-        init(service: UserService) {
+        init(service: NetworkService) {
             self.service = service
         }
         
-        func getUsers(cachedUsers: FetchedResults<User>?,context: NSManagedObjectContext) async {
+        func getUsersAction(isRefresh: Bool = false, cachedUsers: FetchedResults<User>?,context: NSManagedObjectContext) async {
```

Comment:
```suggestion
        func getUsersAction(isRefresh: Bool = false, cachedUsers: FetchedResults<User>?, context: NSManagedObjectContext) async {
```
а ещё строчка слишком длинная, можно каждый параметр на новую строчку перенести